streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -658,7 +658,7 @@ trait ZSink[-R, +E, +A0, -A, +B] extends Serializable { self =>
                 if (self.cont(s2)) UIO.succeedNow(Left(s2))
                 else self.extract(s2).map(Right(_))
               },
-            { case (b, leftover) => UIO.succeedNow(Right((b, leftover ++ Chunk.single(a)))) }
```

Comment:
You can apply `ev` here to resolve the type inference issues.

Reply:
Ok!

```scala
leftover ++ Chunk.single(ev(a))
```

Reply:
Yes. I think the constraints in some of these sink combinators could possibly be simplified further but that is a problem for another day.